src/components/static-map.js

```diff
@@ -182,15 +181,7 @@ export default class StaticMap extends PureComponent {
     const mapStyle = newProps.mapStyle;
     const oldMapStyle = oldProps.mapStyle;
     if (mapStyle !== oldMapStyle) {
-      if (isImmutableMap(mapStyle)) {
-        if (this.props.preventStyleDiffing) {
-          this._map.setStyle(mapStyle.toJS());
-        } else {
-          setDiffStyle(oldMapStyle, mapStyle, this._map);
-        }
-      } else {
-        this._map.setStyle(mapStyle);
-      }
+      this._map.setStyle(normalizeStyle(mapStyle), {diff: !this.props.preventStyleDiffing});
```

Comment:
Is there any way we can automate this `preventStyleDiffing` prop? I mostly use it when I send in a string (url) style, doesn't feel like something the app should need to do in that case.

Reply:
I don't think you need to set this prop with string urls. If so it is a bug.